shared/src/types/storage.rs

```diff
@@ -202,6 +202,21 @@ impl Key {
         Ok(Key { segments })
     }
 
+    /// Returns the last segmenet without popping it
+    pub fn peek(&self) -> Option<&DbKeySeg> {
```

Comment:
it looks like these are not used, but if we do add them, I think `peek` should be called `last` (same as the method on Vec, which we can also call to implement it) and pop might be better as `pop(&mut self) -> Result<DBKeySeg>`

Reply:
Ill remove them